go-controller/pkg/crd/egressfirewall/v1/types.go

```diff
@@ -0,0 +1,75 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EgressNetworkFirewallRuleType indicates whether an EgressNetworkFirewallRule allows or denies traffic
+type EgressFirewallRuleType string
+
+const (
+	EgressFirewallRuleAllow EgressFirewallRuleType = "Allow"
+	EgressFirewallRuleDeny  EgressFirewallRuleType = "Deny"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=egressfirewall
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EgressFirewall describes the current egress firewall for a Namespace.
+// Traffic from a pod to an IP address outside the cluster will be checked against
+// each EgressFirewallRule in the pod's namespace's EgressFirewall, in
+// order. If no rule matches (or no EgressFirewall is present) then the traffic
+// will be allowed by default.
+type EgressFirewall struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata, omitempty"`
+	// Specification of the desired behavior of EgressFirewall.
+	Spec EgressFirewallSpec `json:"spec"`
+	// Observed status of EgressFirewall
+	Status EgressFirewallStatus `json:"status"`
+}
+
+// EgressFirewallStatus describes the lifecycle status of EgressFirewall.
+type EgressFirewallStatus struct {
+	State   string `json:"state"`
+	Message string `json:"message"`
+}
+
+// EgressFirewallSpec is a desired state description of EgressFirewall.
+type EgressFirewallSpec struct {
+	// a collection of egress firewall rule objects
+	EgressFirewallRules []EgressFirewallRule `json:"egressfirewallrules"`
```

Comment:
just `Rules` is fine. This is the `EgressFirewall`. You know that they're egress firewall rules.

(For the field name; the type name should stay redundant.)

Reply:
changed the EgressFirewallRules -> Rules and EgressFirewallRule -> Rule

Reply:
sorry, I meant to change the field name but keep the type name. `Rules []EgressFirewallRule`

Reply:
sorry Dan, I think I got it now the field name is rules but the types are EgressFirewallRule and EgressFirewallRules